File: PYTHON/CheckBinaryStringOneSegment.py
"""
1784. Check if Binary String Has at Most One Segment of Ones

Problem Statement:
You are given a binary string s containing only '0' and '1'.
The string has no leading zeros, meaning it always starts with '1'.

Your task is to check whether the string contains at most
one contiguous segment of '1's.

A segment means a continuous block of '1's.

Example 1
Input: "1001"
Output: False
Explanation:
There are two separate segments of '1':
"1" at the start and another "1" at the end.

Example 2
Input: "110"
Output: True
Explanation:
All '1's appear in a single block.

Key Observation:
Because the string always starts with '1', the first segment is guaranteed.

If another segment of '1' appears later, we must have seen:
0 -> 1 transition

That pattern is represented by substring "01".

So if "01" exists in the string, there are multiple segments.
Otherwise, there is only one segment.

Constraints:
1 <= len(s) <= 100
s contains only '0' and '1'
s[0] is always '1'
"""

def checkOnesSegment(s):

    # If "01" exists, it means after some zero,
    # a new '1' started again → second segment.
    return "01" not in s


# User input
s = input("Enter binary string: ")

result = checkOnesSegment(s)

print("At most one segment of ones:", result)

File: JAVA/Leetcode/CheckBinaryStringOneSegment.java
/*
1784. Check if Binary String Has at Most One Segment of Ones

Problem Statement:
You are given a binary string s consisting only of characters '0' and '1'.
The string does NOT contain leading zeros, meaning it always starts with '1'.

Your task is to determine whether the string contains at most ONE contiguous
segment of '1's.

A contiguous segment means the '1's appear together without interruption.

Examples:

Example 1
Input: s = "1001"
Output: false
Explanation:
There are two separate segments of '1':
"1" at the beginning and another "1" at the end.

Example 2
Input: s = "110"
Output: true
Explanation:
The ones appear only once in a single contiguous block.

Key Idea:
Since the string starts with '1', the first segment of ones is guaranteed.
A second segment would only occur if we see the pattern "01"
(which means we had 1 -> 0 -> 1 transition).

So if the substring "01" exists, it means a new segment of ones begins later.

Constraints:
1 <= s.length <= 100
s[i] is either '0' or '1'
s[0] is '1'
*/

import java.util.Scanner;

public class CheckBinaryStringOneSegment {

    public static boolean checkOnesSegment(String s) {

        /*
        If "01" appears in the string, it means:
        first we had a '0', then a '1' started again,
        meaning a new segment of ones began.

        That would violate the rule of having only one segment.
        */

        return !s.contains("01");
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        // Taking binary string input from user
        System.out.print("Enter binary string: ");
        String s = sc.next();

        boolean result = checkOnesSegment(s);

        System.out.println("At most one segment of ones: " + result);

        // Close scanner to avoid resource leaks
        sc.close();
    }
}